app/Http/Controllers/Api/V1/PaletteController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\StorePaletteRequest;
+use App\Http\Requests\Api\UpdatePaletteRequest;
+use App\Http\Resources\PaletteIndexResource;
+use App\Http\Resources\StorePaletteResource;
+use App\Models\Palette;
+use App\Services\ApiResponse\ApiResponseFacade;
+use App\Services\Palette\PaletteService;
+use Illuminate\Support\Facades\Gate;
+
+class PaletteController extends Controller
+{
+    public function __construct(private PaletteService $paletteService)
+    {
+    }
+
+    public function index()
+    {
+        $palettes = Palette::with('user')->paginate(10);
+        return ApiResponseFacade::withSuccess(true)
+            ->withStatus(200)
+            ->withData(new PaletteIndexResource($palettes))
+            ->withMessage('')
+            ->build()->response();
+    }
+
+    public function store(StorePaletteRequest $request)
+    {
+        $newPalette = $this->paletteService->store($request->input('colors'));
+
+        return ApiResponseFacade::withSuccess($newPalette->isSuccess())
+            ->withMessage($newPalette->getMessage())
+            ->withData(new StorePaletteResource($newPalette->getData()))
+            ->withStatus(200)
+            ->build()->response();
+    }
+
+    public function update(UpdatePaletteRequest $request, Palette $palette)
+    {
+        Gate::authorize('update', $palette);
+        $updatedPalette = $this->paletteService->update($palette, $request->input('colors'));
+
+        return ApiResponseFacade::withSuccess($updatedPalette->isSuccess())
+            ->withStatus(200)
```

Comment:
`200` status code is default behavior of this `ApiResponseFacade` so we don't need to set it again